suse_migration_services/units/product_setup.py

```diff
@@ -58,29 +60,74 @@ def main():
         products_metadata = os.sep.join(
             [root_path, 'etc', 'products.d']
         )
-        baseproduct = os.sep.join(
-            [products_metadata, 'baseproduct']
-        )
-        if os.path.exists(baseproduct):
+        if os.path.exists(products_metadata):
             log.info('Creating backup of Product data')
             Command.run(
                 [
                     'rsync', '-zav', '--delete', products_metadata + os.sep,
                     '/tmp/products.d.backup/'
                 ]
             )
-            log.info('Updating Base Product to be suitable for migration')
-            xml = ElementTree()
-            xml.parse(baseproduct)
+        baseproduct = get_baseproduct(products_metadata)
+        if baseproduct.count != 1:
+            if not baseproduct.count:
+                message = 'There is no baseproduct'
+            else:
+                message = ('Found multiple product definitions '
+                           'without element <flavor>: \n{0}'
+                           .format('\n'.join(baseproduct.prod_filenames)))
+            log.error(message)
+            raise DistMigrationProductSetupException(message)
+
+        log.info('Updating Base Product to be suitable for migration')
+        delete_target_registration(baseproduct.prod_filenames.pop())
```

Comment:
And sorry there is now another issue which came to my mind. Due to Roberts last objection the get_products method was changed to store all base product names as a list. Doing so invalidates the idea of the tuple type. In this case get_products can simply return the list of names. len(list) will be the same as count:

```python
def get_baseproduct(products_metadata):
    base_product_files = []
    prod_filenames = glob.glob(
        os.path.join(products_metadata, '*.prod')
    )
    xml = ElementTree()
    for prod_filename in prod_filenames:
        try:
            xml.parse(prod_filename)
            register_sections = xml.findall('register')
            for register in register_sections:
                flavor = register.findall('flavor')
                if not flavor:
                    base_product_files.append(prod_filename)
        except Exception as issue:
            log.error(
                XML Parsing {0} failed with: {1}'.format(prod_filename, issue)
            )
    return base_product_files
```

Later on this could be used like this

```python
baseproducts = get_baseproduct(products_metadata)
if baseproducts:
    count = len(baseproducts)
    ...
```